### PR TITLE
emu/video.cpp: Set file names via -snapname when exiting in -seconds_…

### PR DESCRIPTION
**Pull request by 0kmg**
…to_run.

### CODE REVIEW

**Comment by 0kmg** [comment]
I dunno if there's a reason this is hard-coded or if someone out there really prefers this behavior? It seemed rather inconvenient when I tried to do some scripted testing on the command line, hence this PR.

**Comment by ajrhacker** [comment]
The regrep utility does depend on these files being named `final.png`. I suppose that can be worked around, especially since this utility is probably used very little except by certain MAME developers for internal testing purposes.

**Comment by 0kmg** [comment]
Well, I don't need this change, just thought it seemed like an obvious plus. I'm not familiar with regrep, but the old behavior for snapshots could be had by doing `-snapname %g/final.png`, I believe. I don't know how much annoyance this would cause Tafoid et al, though.

**Comment by Tafoid** [comment]
I'm all for it provided it works properly and allows full control.  I've been looking for similar function for many years now.
https://mametesters.org/view.php?id=4967

**Review by Tafoid — COMMENTED** [review]
All my personal testing tools are external to the project so I'm unsure how this effects things internally.  It can't hurt so long as the default "final.png" behavior is kept in place leaving the -snapname adjustments up to the user.

**Comment by 0kmg** [comment]
@Tafoid As you can see this is just a 1-line patch. It turns on the exact same file name processing that's used for snapshots and recordings, so it responds to the global `snapname` option whether it came from the command line or an .ini file. As it stands you'd need an .ini file with `snapname %g/final.png` or you'd need to specify `-snapname %g/final.png` to get the previous default behavior.

If I understand your concern, it sounds like you'd want it to respond only to `-snapname` on the command line and ignore `snapname` in .ini files? At least that's what it would mean to default to "final.png", since `snapname` is global and set to `%g/%i` in the absence of any other definition. IMO, it's not the most aesthetically appealing design choice to respond differently based on how the option was set. However, I defer to more expert advice on that and whether there are any other internal concerns.

File: docs/source/commandline/commandline-all.rst [hunk 1]
```diff
@@ -1764,8 +1764,8 @@ Core Performance Options
     benchmarking and automated testing.  By combining this with a fixed set of
     other command line options, you can set up a consistent environment for
     benchmarking MAME's emulation performance.  In addition, upon exit, the
-    **-str** option will write a screenshot called ``final.png`` to the system's
-    snapshot directory.
+    **-str** option will write a screenshot to the system's snapshot directory
+    with the file name determined by **-snapname**.
```
Comment (cuavas) [inline docs/source/commandline/commandline-all.rst:1768]: Should be “…determined by the **-snapname** option.”

**Comment by cuavas** [comment]
In principle I think this is pretty harmless – as @0kmg says, it’s pretty easy to get the previous behaviour using `-snapname` along with `-str`.

File: docs/source/commandline/commandline-all.rst (hunk 1 above)
Comment (0kmg) [reply docs/source/commandline/commandline-all.rst]: Thanks, updated.

**Comment by firewave** [comment]
If you use `-str` and want to perform manual screenshots as well as having the "final" snap this would no longer work since they would all go to the same file - but that obviously doesn't seem to be a likely use case at all. Beside that I see no problems with this change.

Since it's just a parameter change I can handle this in `mame_regtest`. Also with an option to handle old the old behavior. It's obviously not the first time this had to be done.

BTW it would be great if somebody could rebase the mamedev fork of it. Thanks.

**Comment by 0kmg** [comment]
@firewave Ah, I see. Maybe that was what Tafoid was getting at when he said default "final.png" behavior. Yes, now that this change went it, it's true that you can't use a constant name, e.g. `-snapname %g/final.png`, AND take manual screenshots. You'll definitely have to use the incrementing `%i` in there somewhere. I didn't think about that when scratching this itch because I was scripting test screenshots.

**Comment by Tafoid** [comment]
I'd humbly suggest retaining legacy behavior here so that use for -str/-bench will continue to use the final.png and only change if a non-default snapname value is applied.  As of now, when I run with -str/-bench, it saves it as if it were F12'd ... showing 0000.png, etc.  Unless there is a better way to handle it..

**Comment by smf-** [comment]
Pass |-snapname %g/final.png as well when using -str/-bench?
|

**Comment by Tafoid** [comment]
Should be able to - at least in isolated testing I was able to.  Not sure why my regression test is still producing 0000.png though - maybe it is a command order issue.

**Pull request merged**